partiql-parser/src/lalr/mod.rs

```diff
@@ -409,44 +411,63 @@ mod tests {
             assert!(res.is_err());
             let errors = res.unwrap_err();
             assert_eq!(2, errors.len());
-            assert_eq!("Unexpected token [At] at [LineAndCharPosition { line: LineOffset(0), char: CharOffset(21) }..LineAndCharPosition { line: LineOffset(0), char: CharOffset(23) }]", errors[0].to_string());
-            assert_eq!("Unexpected token [At] at [LineAndCharPosition { line: LineOffset(0), char: CharOffset(44) }..LineAndCharPosition { line: LineOffset(0), char: CharOffset(46) }]", errors[1].to_string());
-            assert!(matches!(
+            assert_eq!(
+                "Unexpected token `AT` at `(1:22..1:24)`",
+                errors[0].to_string()
+            );
+            assert_eq!(
+                "Unexpected token `AT` at `(1:45..1:47)`",
+                errors[1].to_string()
+            );
+            assert_eq!(
                 errors[0],
                 ParserError::UnexpectedToken(UnexpectedToken {
                     inner: UnexpectedTokenData {
-                        token: lexer::Token::At
+                        token: Cow::from("AT")
                     },
-                    location: std::ops::Range {
+                    location: Location {
                         start: LineAndCharPosition {
                             line: LineOffset(0),
                             char: CharOffset(21)
-                        },
+                        }
+                        .into(),
                         end: LineAndCharPosition {
                             line: LineOffset(0),
                             char: CharOffset(23)
-                        },
+                        }
+                        .into(),
                     },
                 })
-            ));
-            assert!(matches!(
+            );
+            assert_eq!(
                 errors[1],
                 ParserError::UnexpectedToken(UnexpectedToken {
                     inner: UnexpectedTokenData {
-                        token: lexer::Token::At
+                        token: Cow::from("AT")
                     },
-                    location: std::ops::Range {
+                    location: Location {
                         start: LineAndCharPosition {
                             line: LineOffset(0),
                             char: CharOffset(44)
-                        },
+                        }
+                        .into(),
                         end: LineAndCharPosition {
                             line: LineOffset(0),
                             char: CharOffset(46)
-                        },
+                        }
+                        .into(),
                     },
                 })
-            ));
+            );
+        }
+
+        #[test]
+        fn eof() {
+            let res = parse_partiql(r#"SELECT"#);
```

Comment:
Interestingly, PostgreSQL doesn't give an error for this query:

```
postgres=# SELECT;
--
(1 row)
```
While MySQL and SQLite give an error. I think it makes sense to give an error in this case

Reply:
I suppose you could interpret `SELECT` as the zero-column projection from the zero-row collection. Outside of conceptual completeness, I can't think of the utility of this now -- and we could always add it later.